### PR TITLE
Type check when comparing nodes

### LINKED ISSUE #95: Node __eq__ method internal error
`Node`'s `__eq__` method doesn't check the type of the object it's being compared to, which leads to an exception if you try to compare Nodes to non-nodes.  For example, I can do `5 == "hello"` in Python, but if I try `5 == grammar.parse(text)`, I get the following exception:

```
  File "/Users/wiseman/Dropbox/Disney/src/cca_dialogengine/env/lib/python2.7/site-packages/parsimonious/nodes.py", line 90, in __eq__
    self.expr_name == other.expr_name and
AttributeError: 'int' object has no attribute 'expr_name'
```

### PR DESCRIPTION
**Pull request by cknv**
This should close #95 by making the `Node` give up comparisons with things that are not of its class.

I went with returning `NotImplemented` as that gives `other` a chance to do the comparison.

I thought about using
```python
if not isinstance(other, type(self)):
    ...
```
But I don't really know what is the preferred style, so I just went with the simpler.

### CODE REVIEW

File: parsimonious/nodes.py [hunk 1]
```diff
@@ -90,6 +90,9 @@ def __str__(self):
 
     def __eq__(self, other):
         """Support by-value deep comparison with other nodes for testing."""
+        if not isinstance(other, Node):
+            return NotImplemented
+
         return (other is not None and
```
Comment (erikrose) [inline parsimonious/nodes.py]: Since we're doing the isinstance() test above, we can lose the `other is not None` clause.

File: parsimonious/tests/test_nodes.py [hunk 2]
```diff
@@ -142,3 +142,8 @@ def visit_greeting(self, thing, visited_children):
             raise PrimalScream('This should percolate up!')
 
     assert_raises(PrimalScream, Screamer().parse, 'howdy')
+
+
+def test_node_inequality():
+    node = Node('text', 'o hai', 0, 5)
+    assert node != 5
```
Comment (erikrose) [inline parsimonious/tests/test_nodes.py]: For consistency and to work properly under `python -O` (rather than getting optimized out), let's import `ok_` from `nose.tools` and do `ok_(node != 5)`. Since we're removing the explicit `None` checking above, it might also not be a bad idea to test equality against `None` if we're not doing it elsewhere.

Comment (cknv) [inline parsimonious/tests/test_nodes.py]: Absolutely! I am mostly familiar with pytest and not much with nose, so I just hacked away at it until I had a failing test.

**Comment by erikrose** [comment]
Thanks again for the good patch! :-D

**Comment by wiseman** [comment]
I didn't know about the `NotImplemented` technique, nice.
